reservant-frontend/src/components/RegisterPage/RegisterForm/RegisterForm.tsx

```diff
@@ -0,0 +1,115 @@
+import React, {useState, useEffect} from "react"
+import RegisterValidator from "../RegisterValidator/RegisterValidator";
+
+interface UserProps {
+    name: string,
+    surname: string,
+    password: string,
+    birthDate: string,
+    email: string,
+    phoneNumber: string
+}
+
+interface RegisterFormProps {
+    onSubmit: (user: UserProps, link: string) => void
+}
+
+export default function RegisterForm ({onSubmit}: RegisterFormProps) {
+    
+    const [name, setName] = useState<string>("");
+    const [surname, setSurname] = useState<string>("");
+    const [birthDate, setBirthDate] = useState<string>("");
+    const [email, setEmail] = useState<string>("");
+    const [phoneNumber, setPhoneNumber] = useState<string>("");
+    const [password, setPassword] = useState<string>("");
+    const [passwordRepeated, setPasswordRepeated] = useState<string>("");
+
+    //alternatywnie string z komunikatem co do poprawy, prawdopodobnie bez useEffect wtedy
+    const [message, setMessage] = useState<boolean>(false);
+
+    //funkcja odpowiadająca za przesłanie danych rejestrowanego klienta
+    const submitUser = () => {
+        onSubmit({
+            name: name,
+            surname: surname,
+            password: password,
+            birthDate: birthDate,
+            email: email,
+            phoneNumber: phoneNumber
+        }, 'http://172.21.40.127/POST/auth/register-customer');
+    }
+
+    //funkcja odpowiadająca za przesłanie danych rejestrowanego właściciela
+    const submitOwner = () => {
+        onSubmit({
+            name: name,
+            surname: surname,
+            password: password,
+            birthDate: birthDate,
+            email: email,
+            phoneNumber: phoneNumber
+        }, 'http://172.21.40.127/POST/auth/register-restaurant-owner');
+    }
+
+    //w tym miejscu sprawdzamy przy każdej zmianie zawartości wylistowanych pól, czy wstępna walidacja została spełniona
+    useEffect(() => {
+        if(name.trim().length===0){
+            setMessage(false)
+        }
+        else if(surname.trim().length===0){
+            setMessage(false)
+        }
+        else if(birthDate.length===0){
+            setMessage(false)
+        }
+        //nie wiem czy tu trim potrzebny, zostawiam żeby nie było przypału
+        else if(email.trim().length===0){
```

Comment:
brak validacji hasła przez regexa

Reply:
email jest podawany w input typu email wiec powinno (chyba) robic to za nas

Reply:
no tak ale to jest poleganie na api przeglądarki